### ISSUE TITLE
Edit card on single click, center card on double click or ctrl+ single click

### ISSUE THREAD

**Issue by lifehaschanged** [ycnmhd/obsidian-lineage#57]
### What would you like to be added?

If a card is further away and you click on it only once, the view shifts, i.e. the card is centered. However, this then becomes a problem when you want to double-click the card with the mouse. Because the first click only centers the card, and then you have to move the mouse back to the card and then you can perform a double-click on the card to edit it. That's why I would suggest two possible solutions here:
1. Trigger the centering mode of a card with the mouse only when double-clicking, with a single click the card will not be centered.
2. Keep the centering mode of a card with a single click, but allow an additional modifier, e.g. alt+click or ctrl+click on a card. With such a modifier click, a double-click would then be simulated, which means the card would be centered but would also go into edit mode at the same time.

### Why is this feature needed?

It would facilitate the editing of cards.

### Estimated usefulness for plugin users

Useful for many users

**Comment by ycnmhd** [comment]
You can already do this by holding `ctrl` (or `cmd`):
- `ctrl` + `left click` to make a card active without centering it
- `ctrl` + `double click` to enable editing and center the card

**Comment by lifehaschanged** [comment]
Thank you, I didn't know that. It would be wonderful if there was a new setting for mouse usage for editing cards. For example, there could be an option for:

1. Single click > When selected, editing the card is already enabled with a simple mouse click and the card is centered (those who only want to select and center the card as it is now could do so with ctrl/cmd + click)

2. Double click > The current already implemented behavior is activated

But if that doesn't suit you, it would also be helpful for me if there was an option to set the card centering mode. Options could be:

1 > Card is centered on simple mouse click
2 > Card is centered on double click  
3 > Disable card centering

Because then I could double-click on the card and edit it without having to press a modifier. This would be helpful for people like me who often need to go through the text with the mouse.

**Comment by ycnmhd** [comment]
3 > Disable card centering

You can set `scrolling behavior` to `reveal active card` to only scroll the document horizontally when necessary.
I experimented with disabling vertical centering of the active branch, but I found it impractical

**Comment by lifehaschanged** [comment]
I understand and thank you for the clarification regarding the centering of the cards.
It would of course be great if you could one day implement editing the cards with just a single mouse click. I appreciate it.

**Comment by ycnmhd** [comment]
Version 0.8.0 includes some improvements related to your request:  
- A setting to keep edit mode active when another card is selected.  
- The editor-state condition for hotkeys is now configurable, allowing navigation hotkeys to be used while editing (e.g., `arrow keys` only when the editor is inactive, and `arrow keys + modifiers` when the editor is active).  
- A "lazy vertical scrolling mode" that scrolls just enough to reveal the active branch (e.g., it won’t scroll the parent when a sibling of the active card is selected).

**Comment by lifehaschanged** [comment]
Wow, great new things you implementeted there! 
Really cool the maintain edit mode setting (single clicks, thanks!), the outline (great), drag to pan (thanks, works good with the space key), pin and edit cards in the sidebar plus card history (good idea! small observation: I think mouse scrolling is not working in the sidebar but navigating with keys works + the buttons on top are missing when navigating back on first card inside the sidebar), the mini map (great for quick navigating also) and the shorctuts, lazy vertical scroll mode.
Big thanks and much appreciated!

**Comment by ycnmhd** [comment]
> I think mouse scrolling is not working in the sidebar but navigating with keys works + the buttons on top are missing when navigating back on first card inside the sidebar)

Thank you for pointing this out! I did not notice this regression.
